File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    let answer = [];
    let lastIndex = {}; // 문자별 마지막 등장 위치 저장

    for (let i = 0; i < s.length; i++) {
        const char = s[i];

        if (lastIndex.hasOwnProperty(char)) {
            answer.push(i - lastIndex[char]);
        } else {
            answer.push(-1);
        }

        lastIndex[char] = i; // 현재 위치 갱신
    }

    return answer;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    last_index = {}  # 문자의 마지막 등장 위치를 저장하는 딕셔너리

    for i, c in enumerate(s):
        if c in last_index:
            answer.append(i - last_index[c])
        else:
            answer.append(-1)
        last_index[c] = i  # 현재 위치로 갱신

    return answer
